lib/internal/fs/streams.js

```diff
@@ -69,17 +126,36 @@ function ReadStream(path, options) {
     options.emitClose = false;
   }
 
+  options.autoDestroy = options.autoClose === undefined ?
+    true : options.autoClose;
+  options.path = path;
   Readable.call(this, options);
+}
+ObjectSetPrototypeOf(ReadStream.prototype, Readable.prototype);
+ObjectSetPrototypeOf(ReadStream, Readable);
+
+ObjectDefineProperty(ReadStream.prototype, 'autoClose', {
+  get() {
+    return this._readableState.autoDestroy;
+  },
+  set(val) {
+    this._readableState.autoDestroy = val;
+  }
+});
+
+const openReadFs = internalUtil.deprecate(function() {
+}, 'ReadStream.prototype.open() is deprecated', 'DEP0135');
```

Comment:
Nit: perhaps add a comment inside that noop is intentional? (and in the `Writable` too)